### PR TITLE
SDN docs and stuff

### PR DESCRIPTION
**Pull request by danwinship**
follow-ups from review comments on #9227:

- `oc describe` didn't support the SDN types
- swagger docs should use the json/yaml field names, not the go field names
- the docs weren't entirely useful/accurate/up-to-date

@openshift/networking PTAL

### CODE REVIEW

File: pkg/sdn/api/v1/types.go [hunk 1]
```diff
@@ -11,19 +11,21 @@ const (
...
+// ClusterNetwork describes the cluster network. There is normally only one object of this type,
+// named "default", which is created by the SDN network plugin based on the master configuration
+// when the cluster is brought up for the first time.
 type ClusterNetwork struct {
 	unversioned.TypeMeta `json:",inline"`
 	// Standard object's metadata.
 	kapi.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
-	// Network is a CIDR string to specify the global overlay network's L3 space
+	// network is a CIDR string specifying the global overlay network's L3 space
```
Comment (stevekuznetsov) [inline pkg/sdn/api/v1/types.go]: I think we want to keep these as uppercase to be valid Godoc as that's more important than it being consistent with the generated swaggerdoc for the json field name

**Comment by dcbw** [comment]
Other than comments @stevekuznetsov , LGTM.

**Comment by pravisankar** [comment]
LGTM

File: pkg/sdn/api/v1/types.go (hunk 1 above)
Comment (danwinship) [inline pkg/sdn/api/v1/types.go]: @stevekuznetsov @smarterclayton explicitly requested lowercase field names in docs in https://github.com/openshift/origin/pull/9227#discussion_r71911568. (Though it seems that only pkg/build currently follows that convention.)

**Comment by smarterclayton** [comment]
[test]

File: pkg/sdn/api/v1/types.go (hunk 1 above)
Comment (stevekuznetsov) [inline pkg/sdn/api/v1/types.go]: @smarterclayton that's surprising ... why do we want this?

Comment (smarterclayton) [inline pkg/sdn/api/v1/types.go]: This Godoc is turned into swagger, which is read by humans in the context of JSON responses.   Someone reading swagger will never see the Go types, they will only ever see the JSON responses.  Returning the Go type name is thus not useful to them, while the JSON field name is.

Comment (stevekuznetsov) [inline pkg/sdn/api/v1/types.go]: Right, but why don't we want Godoc to read correctly for developers using the lib? I think the correct approach here would be to edit the comments programmatically when we generate Swagger descriptions from them. It doesn't feel right to break Godoc conventions to appease some code generator downstream.

**Comment by danwinship** [comment]
Repushed without the uppercase/lowercase changes. Should I file a bug about making the doc generation process fix up the field names automatically?

**Comment by stevekuznetsov** [comment]
@danwinship sure, the tool lives upstream so file it against k8s please

**Comment by danwinship** [comment]
@stevekuznetsov done. ok to merge?

**Comment by stevekuznetsov** [comment]
LGTM, has @smarterclayton signed off?

**Comment by smarterclayton** [comment]
Looks good to me.

**Comment by stevekuznetsov** [comment]
[merge]

**Pull request merged**